lib/browser.js

```diff
@@ -307,13 +307,14 @@ Browser.prototype.close = async () => {
   try {
     // Allow the beforeunload event to be executed
     // https://github.com/puppeteer/puppeteer/blob/v1.11.0/docs/api.md#pagecloseoptions
-    if (this.page) await this.page.close({ runBeforeUnload: true });
+    if (this.page && !this.page.isClosed())
+      await this.page.close({ runBeforeUnload: true });
 
     // The page is closed, let's close the browser
-    if (this.browser) await this.browser.close();
+    if (this.browser && this.browser.isConnected()) await this.browser.close();
   } catch (ex) {
     debug("An exception was raised in Browser.prototype.close(): " + ex);
-    throw ex;
+    debug(ex);
```

Comment:
For several reasons, sometimes the browser is already closed and closing it again fails.

One of the reasons is if the browser crashes. It occurs sometimes.

Another reason is when the main request responds with an HTTP error code (404, 500...). The code inside /lib/index.js closes the browser. Then later, phantomas tries to close it a second time.